Lib/enum.py

```diff
@@ -104,10 +101,12 @@ def __setitem__(self, key, value):
                 # enum overwriting a descriptor?
                 raise TypeError('%r already defined as: %r' % (key, self[key]))
             if isinstance(value, auto):
-                self._auto_called = True
-                if value.value == _auto_null:
-                    value.value = self._generate_next_value(key, 1, len(self._member_names), self._last_values[:])
-                value = value.value
+                _value = value.value
+                if _value == _auto_null and self._auto_called:
```

Comment:
`self._auto_called` is never set to true inside `_EnumDict`, so this test will never succeed.